app/controllers/password_reset_controller.rb

```diff
@@ -0,0 +1,28 @@
+class PasswordResetController < ApplicationController
+
+  def create
+    user = User.find_by(email: params[:email])
+    user.send_password_reset if user
+    redirect_to login_url, :notice => "Email sent with password reset instructions."
+  end
+
+  def edit
+    @user = User.find_by(password_reset_token: params[:id])
+    if @user.password_reset_token_generated_at < 2.hours.ago
+      redirect_to new_password_reset_path, :alert => "Password reset has expired."
+    end
+  end
+
+  def update
+    @user = User.find_by(password_reset_token: params[:id])
+    if @user.update_attributes(user_params)
```

Comment:
@rishabhguptarishi 
Don't use update_attributes here, update_attributes updates all attributes passed to it. Here we want to update password only. You have handelled it by user_params method by permit, but just in case if in future we add something in permit, that attribute will also be updated. So lets manually set password, confirm_password to avoid such case